vm/src/vm_hooks/vh_handler/vh_blockchain.rs

```diff
@@ -35,6 +35,10 @@ pub trait VMHooksBlockchain: VMHooksHandlerSource {
         );
     }
 
+    fn get_shard_of_address(&self, address_bytes: &[u8]) -> i32 {
```

Comment:
You also need some glue code in vh_dispatcher.rs, I believe that is also missing.